ecosystem/sf-indexer/aptos-sf-indexer/src/database.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+//! Database-related functions
+#![allow(clippy::extra_unused_lifetimes)]
+use std::{cmp::min, sync::Arc};
+
+use diesel::{
+    pg::PgConnection,
+    r2d2::{ConnectionManager, PoolError, PooledConnection},
+    RunQueryDsl,
+};
+
+pub type PgPool = diesel::r2d2::Pool<ConnectionManager<PgConnection>>;
+pub type PgDbPool = Arc<PgPool>;
+pub type PgPoolConnection = PooledConnection<ConnectionManager<PgConnection>>;
+
+pub const MAX_DIESEL_PARAM_SIZE: u16 = u16::MAX;
+
+/// Given diesel has a limit of how many parameters can be inserted in a single operation (u16::MAX)
+/// we may need to chunk an array of items based on how many columns are in the table.
+/// This function returns boundaries of chunks in the form of (start_index, end_index)
+pub fn get_chunks(num_items_to_insert: usize, column_count: usize) -> Vec<(usize, usize)> {
+    let max_item_size = MAX_DIESEL_PARAM_SIZE as usize / column_count;
+    let mut chunk: (usize, usize) = (0, min(num_items_to_insert, max_item_size));
+    let mut chunks = vec![chunk];
+    while chunk.1 != num_items_to_insert {
+        chunk = (
+            chunk.0 + max_item_size,
+            min(num_items_to_insert, chunk.1 + max_item_size),
+        );
+        chunks.push(chunk);
+    }
+    chunks
+}
```

Comment:
this is slightly unintelligible, use two vars or a struct or idk man u killing me